### PR TITLE
Sort order

### LINKED ISSUE #5: sort(method = "radix") and order(method = "radix")
only available for R >= 3.3.0, with caveats:

- character vectors are sorted in C locale
- stable, so fallback should use `method = "shell"`
- `method = "radix"` was available for `sort()` under certain conditions in R < 3.3.0

### PR DESCRIPTION
**Pull request by krlmlr**
Fixes #5.

### CODE REVIEW

File: R/sort.R [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+#' @rawNamespace if (getRversion() < "3.3.0") export(sort)
+#' @examples
+#' # get function from namespace instead of possibly getting
+#' # implementation shipped with recent R versions:
+#' bp_sort = getFromNamespace("sort", "backports")
+#'
+#' bp_sort(runif(10), method = "radix")
+sort = function (x, decreasing = FALSE, method = c("shell", "quick", "radix"), ...) {
+  method = match.arg(method)
+  if (identical(method, "radix"))
```
Comment (krlmlr) [inline R/sort.R]: How about temporary changing to the C locale if x is a character vector? Also, if x is an integer, this worked already in some cases for R < 3.3.0.

**Comment by mllg** [comment]
Both functions now use "C" as collate order. But it does not work on R-devel yet.

**Comment by mllg** [comment]
Correction: Fails also with release and 3.3.x

**Comment by mllg** [comment]
Hm, the default for `method` seems to have changed again in r-devel. I start to doubt if this backport is worth the effort. Explicitly setting `method = "radix"` would only have an effect for R-3.3.x (radix sort will be the default for R >3.4.0), and the speedup might not be significant for many applications.

Furthermore, if I have to respect too many corner cases for `sort` it might get much slower for small input `x` in comparison to just doing the shell sort. Your thoughts?

**Comment by krlmlr** [comment]
The pull request looks good already, what's wrong with the tests?

However: A coarse search has found only one package that explicitly uses method = "radix": https://github.com/search?l=R&p=4&q=radix+org%3Acran&type=Code&utf8=%E2%9C%93.

**Comment by mllg** [comment]
I tend to trust the R core team here to select the "best" sorting algorithm and would encourage users to not explicitly "overwrite" the default. The potential gain is relatively small compared to the amount of work one might have to invest to get this right.

If you really want it included in the package, you can of course take over this PR.

**Comment by krlmlr** [comment]
I agree that this might not be as useful as I originally thought. Thanks anyway for looking into it.

**Pull request closed without merging**